gulpfile.babel.js

```diff
@@ -0,0 +1,232 @@
+ */
+gulp.task("clean", (cb) => del(["./dist/", "./docs/"], cb));
+
+
+
+
+/**
+ * =======================================================
+ * $ gulp copy-assets
+ * =======================================================
+ */
+gulp.task("copy-assets", () => {
+  return gulp.src(["./site/assets/**/*"])
+  .pipe(gulp.dest("./dist/assets/"))
+  .pipe(bs.stream());
+});
+
+
+
+/**
+ * =======================================================
+ * $ gulp jade
+ * =======================================================
+ */
+gulp.task("jade", () => {
+  return gulp.src([
+    "./site/jade/**/*.jade",
+    "!./site/jade/**/_*.jade"
+  ])
+  .pipe($.plumber())
+  .pipe($.data(() => {
+    return require("./site/jade/config.json");
+  }))
+  .pipe($.jade({pretty: true}))
+  .pipe(gulp.dest("./docs"))
+  .pipe(bs.stream());
+});
+
+
+
+/**
+ * =======================================================
+ * $ gulp webpack
+ * =======================================================
+ */
+gulp.task("webpack", (cb) => {
+  webpack(webpackConfig, (err, stats) => {
+    if( err ) throw new $.util.PluginError("webpack", err);
+    $.util.log("[webpack]", stats.toString());
+    bs.reload();
+    cb();
+  })
+});
+
+
+
+/**
+ * =======================================================
+ * $ gulp uglify
+ * =======================================================
+ */
+gulp.task("uglify", () => {
+  return gulp.src("./dist/**/*.js")
+  .pipe($.uglify({preserveComments: "some"}))
+  .pipe(gulp.dest("./dist/"))
+  .pipe(bs.stream());
+});
+
+
+
+/**
+ * =======================================================
+ * $ gulp sass
+ * =======================================================
+ */
+gulp.task("sass", () => {
+  let params = {
+    outputStyle: isProduction ? "compressed" : "expanded"
+  };
+
+  return gulp.src("./site/stylesheets/**/*.scss")
+  .pipe($.plumber())
+  .pipe($.sass.sync(params).on("error", $.sass.logError))
+  .pipe($.autoprefixer({
+    browsers: [
+      "last 4 versions",
+      "ie 9",
+      "iOS 6",
+      "Android 4"
+    ]
+  }))
+  .pipe(gulp.dest("./docs/css"))
+  .pipe(bs.stream());
+});
+
+
+
+/**
+ * =======================================================
+ * $ gulp build
+ * =======================================================
+ */
+gulp.task("build", (cb) => {
+  runSequence(
+    "clean",
+    ["webpack", "sass", "jade"],
+    "uglify",
+    cb
+  );
+});
+
+
+
+/**
+ * =======================================================
+ * $ gulp watch
+ * =======================================================
+ */
+gulp.task("watch", (cb) => {
+  runSequence(
+    "build",
+    "bs",
+    () => {
+      $.watch("./site/jade/**/*", () => {
+        gulp.start("jade");
+      });
+
+      // WEBPACK HMR did the job with browser sync
+      $.watch("./site/js/**/*", () => {
+        gulp.start("webpack");
+      });
+
+      $.watch("./site/sass/**/*", () => {
+        gulp.start("sass");
+      });
+
```

Comment:
What about listening for changes on the source file?
```
$.watch("./lib/**/*", () => {
    gulp.start("build");
});
```

Reply:
@tomasmcm everything related to script is done by webpack. `Check webpack.config.babel.js`

Reply:
@renatodeleao that does not work. How about this:
```
$.watch("./lib/**/*", () => {
    gulp.start("webpack");
});
```

Reply:
Sry mate, I was being stupid.

Yes just is like you said above, but passing a stream paths as an array, instead of duplicating the watch function like this:

```javascript
$.watch(["./site/js/**/*", "./lib/**/*" ], () => {
  gulp.start("webpack");
});

Will commit an update. Thanks :thumbs_up: